lightning/src/ln/outbound_payment.rs

```diff
@@ -1192,6 +1197,13 @@ impl OutboundPayments {
 							RetryableSendFailure::OnionPacketSizeExceeded,
 						));
 					}
+
+					// If we expect the HTLCs for this payment to be held at our next-hop counterparty, don't
+					// retry the payment. In future iterations of this feature, we will send this payment via
+					// trampoline and the counterparty will retry on our behalf.
+					if hold_htlcs_at_next_hop {
```

Comment:
And the case where our trampoline-counterparty didn't want to retry, or has some other kind of problem. I think then we ourselves still want to retry with a different peer?

Although it is probably mostly theoretically. In a client<->LSP setup, there isn't any other peer to retry with.